### PR TITLE
9820 add event code and update the form

### PR DESCRIPTION
**Pull request by rosiehsmith**
image
image

### CODE REVIEW

**Review by zachrog — APPROVED** [review]

File: shared/src/business/entities/EntityValidationConstants.js [hunk 1]
```diff
@@ -35,6 +35,11 @@ const DOCKET_ENTRY_VALIDATION_RULE_KEYS = {
   addToCoversheet: joi.boolean().optional(),
   additionalInfo: JoiValidationConstants.STRING.max(500).optional(),
   additionalInfo2: JoiValidationConstants.STRING.max(500).optional(),
+  amicusCuriae: JoiValidationConstants.STRING.when('eventCode', {
+    is: joi.exist().valid('AMBR'),
+    otherwise: joi.optional(),
+    then: joi.required(),
+  }).description('The amicus curiae filing party is required.'),
```
Comment (rachelschneiderman) [inline shared/src/business/entities/EntityValidationConstants.js]: To me, this reads a little like a validation error message. Does this property represent the filing party of an Amicus Brief, if so, could the description be updated to reflect that?

File: shared/src/business/entities/DocketEntry.ts [hunk 2]
```diff
@@ -351,6 +353,11 @@ and contact info from the raw docket entry
         practitioner => practitioner.partyPrivatePractitioner,
       );
 
+      if (this.amicusCuriae) {
```
Comment (rachelschneiderman) [inline shared/src/business/entities/DocketEntry.ts:356]: Out of curiosity, why not add this to the `generateFiledBy` method, just to keep related logic together?

File: web-client/src/presenter/computeds/filingPartiesFormHelper.ts [hunk 3]
```diff
@@ -29,13 +29,16 @@ export const filingPartiesFormHelper = (get, applicationContext) => {
     caseDetail.partyType === PARTY_TYPES.petitionerSpouse ||
     caseDetail.partyType === PARTY_TYPES.petitionerDeceasedSpouse;
 
+  const showFilingPartiesAsCheckboxes = form.eventCode !== 'AMBR';
```
Comment (rachelschneiderman) [inline web-client/src/presenter/computeds/filingPartiesFormHelper.ts:34]: Wonderful name for this property, very clear what it's for 👏 👏

File: shared/src/business/entities/DocketEntry.ts (hunk 2 above)
Comment (rosiehsmith) [reply shared/src/business/entities/DocketEntry.ts:356]: I think it is in that method?

Comment (rachelschneiderman) [reply shared/src/business/entities/DocketEntry.ts:356]: Well that's embarrassing... I will learn how to diff. Looks great!

**Review by rachelschneiderman — APPROVED** [review]

**Pull request merged**